### ISSUE TITLE
[Project Monitoring V2] Rebase Project Monitoring against 40.x.x Rancher Monitoring

### ISSUE THREAD

**Issue by aiyengar2** [rancher/rancher#39724]
Followup for https://github.com/rancher/rancher/issues/39725.

Also introduce e2e CI to validate rebase and introduce documentation and script to make it easier to perform rebases in the future.

**Comment by aiyengar2** [comment]
Need to bump to v0.4.0 of charts-build-scripts for rebase scripts in https://github.com/rancher/prometheus-federator/pull/30

**Comment by aiyengar2** [comment]
Due to changes introduced in `v0.13.0` of https://github.com/k3s-io/helm-controller, the following RKE2 versions need to be verified to ensure they still work in a Project Monitoring V2 environment out-of-the-box.

Produced using the following script on each repo:
```bash
for tag in $(git tag); do
    version=$(git show ${tag}:go.mod 2>/dev/null | grep github.com/k3s-io/helm-controller | cut -d' ' -f2)
    [[ -z ${version} ]] && continue
    [[ ${version} != "v0.13.0" ]] && [[ ${version} != "v0.13.1" ]] && continue
    echo "${tag}: github.com/k3s-io/helm-controller ${version}"
done
```

```
rke2:
v1.23.14+rke2r1: github.com/k3s-io/helm-controller v0.13.0
v1.23.14-rc1+rke2r1: github.com/k3s-io/helm-controller v0.13.0
v1.23.14-rc2+rke2r1: github.com/k3s-io/helm-controller v0.13.0
v1.23.14-rc3+rke2r1: github.com/k3s-io/helm-controller v0.13.0
v1.24.8+rke2r1: github.com/k3s-io/helm-controller v0.13.0
v1.24.8-rc1+rke2r1: github.com/k3s-io/helm-controller v0.13.0
v1.24.8-rc2+rke2r1: github.com/k3s-io/helm-controller v0.13.0
v1.25.4+rke2r1: github.com/k3s-io/helm-controller v0.13.0
v1.25.4-rc1+rke2r1: github.com/k3s-io/helm-controller v0.13.0
v1.25.4-rc2+rke2r1: github.com/k3s-io/helm-controller v0.13.0
```

```
k3s:
v1.23.14+k3s1: github.com/k3s-io/helm-controller v0.13.0
v1.23.14-rc1+k3s1: github.com/k3s-io/helm-controller v0.13.0
v1.23.14-rc2+k3s1: github.com/k3s-io/helm-controller v0.13.0
v1.23.14-rc3+k3s1: github.com/k3s-io/helm-controller v0.13.0
v1.24.8+k3s1: github.com/k3s-io/helm-controller v0.13.0
v1.24.8-rc1+k3s1: github.com/k3s-io/helm-controller v0.13.0
v1.24.8-rc2+k3s1: github.com/k3s-io/helm-controller v0.13.0
v1.24.8-rc3+k3s1: github.com/k3s-io/helm-controller v0.13.0
v1.25.4+k3s1: github.com/k3s-io/helm-controller v0.13.0
v1.25.4-rc1+k3s1: github.com/k3s-io/helm-controller v0.13.0
v1.25.4-rc2+k3s1: github.com/k3s-io/helm-controller v0.13.0
v1.25.4-rc3+k3s1: github.com/k3s-io/helm-controller v0.13.0
v1.25.4-rc4+k3s1: github.com/k3s-io/helm-controller v0.13.0
```

## TLDR for community users seeing errors with these versions

The root reason why you might see this error is because Project Monitoring V2 required a workaround in its initial release to set `helmProjectOperator.helmController.enabled: false` since RKE2 / K3s's Helm Controller operated on a cluster-wide level and didn't respect the managedBy annotation. Now later releases of RKE2 / K3s indicated above do respect that annotation, so the default helm controller can be turned back on.

**Comment by aiyengar2** [comment]
The above note needs to be release noted to let users know that they will need to do this on a Kubernetes version upgrade cc: @btat

**Comment by aiyengar2** [comment]
new chart version will be `1.1.0+up0.2.0-rc1`

We're cutting a new minor since it's a significant bump to the original chart (and the original chart is cutting a minor too)

**Comment by aiyengar2** [comment]
This has been done

**Comment by martyav** [comment]
> The above note needs to be release noted to let users know that they will need to do this on a Kubernetes version upgrade cc: @btat

Does this still need to be noted?
